app/controllers/api/deploy_environments_controller.rb

```diff
@@ -39,4 +42,43 @@ def scale
   def load_more_deployments
     respond_with more_deployments: attributes_for_environment_page(current_user.deploy_environments.find(params[:deploy_environment_id]), params[:page])
   end
+
+  def validate_config_file
+    render status: 422, json: {error: {message: "Schema Validation not supported for given file format. Supported formats: #{SCHEMA_VALIDATION_SUPPORTED_EXT}"}} and return unless schema_validation_supported_format?
+
+    render status: 422, json: {error: {message: "No config_file found matching given path"}} and return unless config_file
+
+    render status: 422, json: {error: {message: "File not found"}} and return unless schema_file_tmp_path
+
+    rx = Rx.new({ :load_core => true })
+    schema = rx.make_schema(YAML.load_file(schema_file_tmp_path))
+    schema.check!(YAML.load(config_file.value))
+  rescue Exception => e
+    render status: 422, json: {error: {message: "Schema validation did not pass. #{e.message}"}}
+  ensure
+    File.delete(schema_file_tmp_path) if schema_file_tmp_path.present? && File.exist?(schema_file_tmp_path)
+  end
+
+  private
+
+  def schema_file_tmp_path
+    return @schema_file_tmp_path if defined? @schema_file_tmp_path
+    @schema_file_tmp_path = "/tmp/#{SecureRandom.uuid}_schema"
```

Comment:
@gja considered creating a tempfile here and found that tempfile can not be created at a specific path. Since here I need to volume mount `/tmp`, decided to keep it as it is.